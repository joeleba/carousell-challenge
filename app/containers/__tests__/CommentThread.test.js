import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import CommentThreadWrapper from '../CommentThread';

const mockStore = configureStore();

const stateNestedComments = {
  contentReducers: {
    currentId: 6,
    posts: {
      0: {
        children: [1, 2],
        content: "ccc",
        downvoteCount: 0,
        upvoteCount: 0,
        title: "sad",
        id: 0
      }
    },
    comments: {
      1: {
        children: [3, 4, 5],
        content: "ccc1",
        downvoteCount: 0,
        upvoteCount: 0,
        id: 1
      },
      2: {
        children: [],
        content: "ccc2",
        downvoteCount: 0,
        upvoteCount: 0,
        id: 2
      },
      3: {
        children: [],
        content: "ccc3",
        downvoteCount: 0,
        upvoteCount: 0,
        id: 3
      },
      4: {
        children: [],
        content: "ccc4",
        downvoteCount: 0,
        upvoteCount: 0,
        id: 4
      },
      5: {
        children: [],
        content: "ccc5",
        downvoteCount: 0,
        upvoteCount: 0,
        id: 5
      },
    },
  },
};

const commentWithChildren = stateNestedComments.contentReducers.comments[1];
const commentWithoutChildren = stateNestedComments.contentReducers.comments[2];

describe('Component: CommentThread', () => {
  it('renders with nested comments', () => {
    const wrapper = shallow(
      <CommentThreadWrapper comment={commentWithChildren} postId={0}/>,
      { context: { store: mockStore(stateNestedComments) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
    expect(wrapper.prop('comments')).toEqual(stateNestedComments['contentReducers']['comments']);
    expect(wrapper.dive().find('FlatList')).toHaveLength(1);
  });

  it('renders with nested comments', () => {
    const wrapper = shallow(
      <CommentThreadWrapper comment={commentWithoutChildren} postId={0}/>,
      { context: { store: mockStore(stateNestedComments) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
    expect(wrapper.prop('comments')).toEqual(stateNestedComments['contentReducers']['comments']);
    expect(wrapper.dive().find('FlatList')).toHaveLength(0);
  });
});
