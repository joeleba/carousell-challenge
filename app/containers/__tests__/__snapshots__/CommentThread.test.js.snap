// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CommentThread renders with nested comments 1`] = `
<View
  style={
    Object {
      "borderColor": "lightgrey",
      "borderLeftWidth": 1,
      "paddingLeft": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     
    ccc1
     
  </Text>
  <Connect(OptionsRow)
    currentComment={
      Object {
        "children": Array [
          3,
          4,
          5,
        ],
        "content": "ccc1",
        "downvoteCount": 0,
        "id": 1,
        "upvoteCount": 0,
      }
    }
    postId={0}
  />
  <FlatList
    data={
      Array [
        Object {
          "children": Array [],
          "content": "ccc3",
          "downvoteCount": 0,
          "id": 3,
          "upvoteCount": 0,
        },
        Object {
          "children": Array [],
          "content": "ccc4",
          "downvoteCount": 0,
          "id": 4,
          "upvoteCount": 0,
        },
        Object {
          "children": Array [],
          "content": "ccc5",
          "downvoteCount": 0,
          "id": 5,
          "upvoteCount": 0,
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`Component: CommentThread renders with nested comments 2`] = `
<View
  style={
    Object {
      "borderColor": "lightgrey",
      "borderLeftWidth": 1,
      "paddingLeft": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     
    ccc2
     
  </Text>
  <Connect(OptionsRow)
    currentComment={
      Object {
        "children": Array [],
        "content": "ccc2",
        "downvoteCount": 0,
        "id": 2,
        "upvoteCount": 0,
      }
    }
    postId={0}
  />
</View>
`;
